### PR TITLE
Add Drift Detection workshop

### PR DESCRIPTION
**Pull request by jamesoff**
*Description of changes:*

Add Drift Detection workshop content.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: workshop/content/intermediate/operations/drift-detection/_index.md [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+{{%expand "Need a hint?" %}}
+* You should delete the resource from the stack, then re-import it again with the `UserData` corrected. You do not need to convert the `UserData` to Base64.
+* Refer to the [Resource Import](resource-importing.html) lab for more guidance.
+{{% /expand %}}
+
+{{%expand "Want to see the solution?" %}}
+
+1. Update the `drift-detection-challenge.yaml` template to add a `DeletionPolicy` attribute with a value of `Retain` to the `Instance1` resource. Save the file.
+2. Update the stack with the updated `drift-detection-challenge.yaml` template. This tells CloudFormation that when the resource is removed from the template, it should not delete it but just stop managing it.
+3. Once the stack is updated, edit the template file again to remove the resource completely (you can comment it out, using the `#` character at the start of each line), and save the file.
```
Comment (mrinaudo-aws) [inline workshop/content/intermediate/operations/drift-detection/_index.md:247]:
```suggestion
3. Once the stack update is complete, edit the template file again to remove the whole resource declaration (you can also choose to comment it out, using the `#` character at the start of each relevant line), and save the file.
```

**Review by mrinaudo-aws — CHANGES_REQUESTED** [review]
Thank you, @jamesoff !  Please review/approve suggestions, and request another review when ready!

File: workshop/content/intermediate/operations/drift-detection/_index.md [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+You can find the template for the solution in `code/solutions/drift-detection/drift-detection-workshop.yaml`.
+
+Well done! You have now learned how to repair drift without impact by deleting and re-importing a resource.
+
+### Cleanup
+
+Follow steps shown next to clean up the resources you created in this workshop.
+
+1. Navigate to the CloudFormation Console.
+2. Choose the stack created in the first lab, for example `drift-detection-workshop`
```
Comment (mrinaudo-aws) [inline workshop/content/intermediate/operations/drift-detection/_index.md:283]:
```suggestion
2. Choose the stack created in the first lab, for example `drift-detection-workshop`.
```

File: workshop/content/intermediate/operations/drift-detection/_index.md [hunk 3]
```diff
@@ -0,0 +1,291 @@
...
+3. Choose **Delete**, then choose the **Delete stack**
```
Comment (mrinaudo-aws) [inline workshop/content/intermediate/operations/drift-detection/_index.md:284]:
```suggestion
3. Choose **Delete**, then choose the **Delete stack**.
```

File: workshop/content/intermediate/operations/drift-detection/_index.md [hunk 4]
```diff
@@ -0,0 +1,291 @@
...
+4. For the `drift-detection-challenge` stack, edit the template file to change the `DeletionPolicy` to `Delete`
```
Comment (mrinaudo-aws) [inline workshop/content/intermediate/operations/drift-detection/_index.md:285]:
```suggestion
4. For the `drift-detection-challenge` stack, edit the template file to change the `DeletionPolicy` to `Delete`.
```

File: workshop/content/intermediate/operations/drift-detection/_index.md [hunk 5]
```diff
@@ -0,0 +1,291 @@
...
+7. Select `Instance1`, then choose **View drift details**.
+8. The drift details show that the `UserData` property has been modified. The `UserData` property is stored using Base64 encoding, so the exact change you made is not obvious in the display.
+
+{{% notice note %}}
+You can use a tool to decode the Base64 text and see the shell script it represents. For example, on Linux you can use the `base64` command line tool to process it as follows.
+Note that some implementations, for example macOS, use `-D` instead of `-d` as the option to the `base64` command.
+{{% /notice %}}
+
+```shell
+$ echo "IyEvdXNyL2Jpbi9lbnYgYmFzaAplY2hvIEhlbGxvIFdvcmxkCg==" | base64 -d
```
Comment (mrinaudo-aws) [inline workshop/content/intermediate/operations/drift-detection/_index.md:227]:
```suggestion
$ echo "IyEvdXNyL2Jpbi9lbnYgYmFzaAplY2hvIEhlbGxvIFdvcmxkCg==" | base64 -d
```
```

File: workshop/content/intermediate/operations/drift-detection/_index.md [hunk 6]
```diff
@@ -0,0 +1,291 @@
...
+5. From **Stack actions**, choose **View drift results**.
+6. The drift status page displays, showing that `Instance1` has been modified.
+7. Select `Instance1`, then choose **View drift details**.
+8. The drift details show that the `UserData` property has been modified. The `UserData` property is stored using Base64 encoding, so the exact change you made is not obvious in the display.
+
+{{% notice note %}}
+You can use a tool to decode the Base64 text and see the shell script it represents. For example, on Linux you can use the `base64` command line tool to process it as follows.
+Note that some implementations, for example macOS, use `-D` instead of `-d` as the option to the `base64` command.
+{{% /notice %}}
+
```
Comment (mrinaudo-aws) [inline workshop/content/intermediate/operations/drift-detection/_index.md:225]:
```suggestion

For example, when you run the following command:

```

File: workshop/content/intermediate/operations/drift-detection/_index.md [hunk 7]
```diff
@@ -0,0 +1,291 @@
...
+```shell
+$ echo "IyEvdXNyL2Jpbi9lbnYgYmFzaAplY2hvIEhlbGxvIFdvcmxkCg==" | base64 -d
+#!/usr/bin/env bash
```
Comment (mrinaudo-aws) [inline workshop/content/intermediate/operations/drift-detection/_index.md:228]:
```suggestion

should display the following output:

```text
#!/usr/bin/env bash
```

**Review by mrinaudo-aws — CHANGES_REQUESTED** [review]
Thank you, @jamesoff !  Can you please review/approve some more suggestions?  Thank you!

**Comment by jamesoff** [comment]
Suggestions applied and tested locally for correct display.

Force-pushed due to rewriting commit to fix author email (using the GitHub UI used my non-Amazon email).

**Review by mrinaudo-aws — APPROVED** [review]
lgtm

**Review by rezabekf — APPROVED** [review]
Approved

**Pull request merged**